src/Stratis.SmartContracts.IntegrationTests/PoW/ContractExecutionFailureTests.cs

```diff
@@ -744,5 +746,40 @@ public void ContractTransaction_Call_Method_Consumes_All_Gas_Throws_OutOfGasExce
             Assert.StartsWith("Execution ran out of gas.", receipt.Error);
             Assert.Equal(preResponse.NewContractAddress, receipt.To);
         }
+
+        [Fact]
+        public void ContractTransaction_Call_Method_Reach_Limit_Of_GasPerBlock_Transaction_NotIncluded_To_Block()
+        {
+            const ulong txGasPrice = 12009;
+            const ulong txGasPerBlockLimit = SmartContractFormatRule.GasLimitMaximum * 10;
+            const int txCount = 100;
+
+            // Ensure fixture is funded.
+            this.node1.MineBlocks(100);
+
+            // Deploy contract.
+            ContractCompilationResult compilationResult = ContractCompiler.CompileFile("SmartContracts/InfiniteLoop.cs");
+            Assert.True(compilationResult.Success);
+
+            var preResponseList = new List<BuildCreateContractTransactionResponse>();
+
+            for (int i = 0; i < txCount; i++)
+            {
+                preResponseList.Add(this.node1.SendCreateContractTransaction(compilationResult.Compilation, 0, gasPrice:10000));
+            }
+            this.node1.WaitMempoolCount(txCount);
+            this.node1.MineBlocks(1);
+
+            NBitcoin.Block lastBlock = this.node1.GetLastBlock();
+
+            int expectedTxQty = Convert.ToInt32(txGasPerBlockLimit / txGasPrice) + 2;
```

Comment:
Where does this + 2 come from?

I think a more obvious way to test this would be to deploy the InfiniteLoop contract in one block - then for the repeated transaction do a `SendCallContractTransaction` that triggers the `Loop` method on the contract - this would consume all of the gas limit per tx always, which is probably more reliable than hardcoding the 12009, which may change if we update our gas costs. Does this make sense @kogot ?

Reply:
@codingupastorm updated. please check.